BaseStation/MainWindow/MapTileDownloadManager.cs

```diff
@@ -0,0 +1,147 @@
+            public Tuple<int, int> ImgDim
+            {
+                get { return _imgDim; }
+                set { if (value.Item1 > 0 && value.Item2 > 0) _imgDim = value; }
+            }
+            private Tuple<double, double> _coords;
+            public Tuple<double, double> Coords
+            {
+                get { return _coords; }
+                set
+                {
+                    if (value.Item1 >= MinLatitude && value.Item1 <= MaxLatitude &&
+                        value.Item2 >= MinLongitude && value.Item2 <= MaxLongitude)
+                        _coords = value;
+                }
+            }
+
+            public Configuration(Tuple<double, double> coords, Tuple<int, int> imgDim, int zoom = 1, int scale = 2, string maptype = "satellite")
+            {
+                Coords = coords;
+                ImgDim = imgDim;
+                Scale = scale;
+                Zoom = zoom;
+                MapType = maptype;
+            }
+
+            // returns the string representation of the configuration
+            public override String ToString()
+            {
+                return "center=" + Coords.Item1 + "," + Coords.Item2 + "&size=" + ImgDim.Item1 + "x"
+                    + ImgDim.Item2 + "&scale=" + Scale + "&zoom=" + Zoom + "&maptype=" + MapType;
+            }
+        }
+
+        // gets the tile set of maps with the given coords of the center, width and height of tiling
+        // and configuration for the center tile
+        public static void DownloadNewTileSet(Tuple<int, int> tilingDim, Configuration config, String mapSetName)
+        {
+            String fileName = Directory.GetCurrentDirectory().ToString() + @"\Images\" + mapSetName + ".evm";
+            using (StreamWriter file = new StreamWriter(fileName))
+            {
+                Tuple<int, int> centerPoint = MapConversion.LatLongToPixelXY(config.Coords.Item1,
+                    config.Coords.Item2, config.Zoom);
+
+                file.WriteLine(config.ImgDim.Item1 + "x" + config.ImgDim.Item2 + "|" + config.Zoom + "|"
+                    + config.Scale + "|" + config.MapType);
+
+                // center of the tiling is 0,0
+                int startx = -tilingDim.Item1 / 2;
+                int starty = -tilingDim.Item2 / 2;
+                if (tilingDim.Item1 % 2 == 0) startx++;
+                if (tilingDim.Item2 % 2 == 0) starty++;
+
+                for (int i = startx; i <= tilingDim.Item1 / 2; i++)
+                {
+                    for (int j = starty; j <= tilingDim.Item2 / 2; j++)
+                    {
+                        Tuple<double, double> newCoords = MapConversion.PixelXYToLatLong
+                            (centerPoint.Item1 + (i * config.ImgDim.Item1), centerPoint.Item2
+                            + (j * config.ImgDim.Item2), config.Zoom);
+                        config.Coords = newCoords;
+                        file.WriteLine(i + "," + j + "|" + Fetch(config));
```

Comment:
Fetch does a fair amount of work, so it would be clearer to see what's going on if it were called on it's own line and not as a part of a larger expression